frontend/src/user/user-editor/user-editor.component.ts

```diff
@@ -127,44 +141,49 @@ export class UserEditorComponent implements OnInit, OnDestroy {
     this.userService.disconnect(this.current.id, form.value.passwordGroup.password)
       .subscribe(data => {
         if (data === true) {
-          this.keep = false;
-          this.deleted = true;
+          this.disconnectFlags.keepLocal = false;
+          this.disconnectFlags.accountDeleted = true;
           this.store.dispatch(new Reload());
         } else {
-          this.disconnectError = data['name'];
+          this.disconnectFlags.error = data['name'];
         }
       });
   }
 
   disconnect_delete() {
-    this.byebye = true;
+    this.disconnectFlags.accountIsBeingDeleted = true;
     this.userService.disconnect_delete(this.current.id)
       .subscribe(data => {
         if (data === true) {
           this.store.dispatch(new AuthActions.Logout());
           this.store.dispatch(new Reload());
         } else {
-          this.disconnectError = data['name'];
+          this.disconnectFlags.error = data['name'];
         }
       });
   }
 
   disconnect_local() {
-    this.userService.disconnect_local().subscribe(data => {
-      if (data === true) {
-        this.disconnected = true;
-        this.store.dispatch(new Reload());
-      } else {
-        this.disconnectError = data['name'];
-      }
-    });
+    this.userService.disconnect_local()
+      .subscribe(data => {
+        if (data === true) {
+          this.disconnectFlags.accountDisconnected = true;
+          this.store.dispatch(new Reload());
+        } else {
+          this.disconnectFlags.error = data['name'];
+        }
+      });
+  }
+
+  activateKeep() {
+    if (!this.disconnectFlags.keepLocal) {
+      this.disconnectFlags.keepLocal = true;
+    }
   }
 
-  changeKeep(i: boolean) {
-    if (i) {
-      this.keep = false;
-    } else {
-      this.keep = true;
+  deactivateKeep() {
+    if (this.disconnectFlags.keepLocal) {
```

Comment:
Amm, dziwnie to wyglada.
this.disconnectFlags.keepLocal = !this.disconnectFlags.keepLocal nie bedzie dzialac?